File: app/src/main/java/com/akay/fitnass/data/db/AppDatabase.java
package com.akay.fitnass.data.db;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;
import android.content.Context;

import com.akay.fitnass.data.db.converter.DateTimeConverter;
import com.akay.fitnass.data.db.converter.ListConverter;
import com.akay.fitnass.data.db.dao.ActiveRunsDao;
import com.akay.fitnass.data.db.dao.RunsDao;
import com.akay.fitnass.data.model.ActiveRuns;
import com.akay.fitnass.data.model.Runs;

@TypeConverters({ListConverter.class, DateTimeConverter.class})
@Database(entities = {Runs.class, ActiveRuns.class}, version = AppDatabase.VERSION, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {
    private static final String DB_NAME = "FitnassDatabase";
    private static AppDatabase mInstance;
    static final int VERSION = 1;

    public abstract RunsDao runsDao();

    public abstract ActiveRunsDao activeRunsDao();

    public static AppDatabase getInstance(final Context context) {
        if (mInstance == null) {
            mInstance = Room.databaseBuilder(context, AppDatabase.class, DB_NAME)
                    .fallbackToDestructiveMigration()
                    .allowMainThreadQueries()
                    .build();
        }
        return mInstance;
    }
}

File: app/src/main/java/com/akay/fitnass/data/db/AppDatabase.kt
package com.akay.fitnass.data.db

import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import android.content.Context

import com.akay.fitnass.data.db.converter.DateTimeConverter
import com.akay.fitnass.data.db.converter.ListConverter
import com.akay.fitnass.data.db.dao.ActiveRunsDao
import com.akay.fitnass.data.db.dao.RunsDao
import com.akay.fitnass.data.model.ActiveRuns
import com.akay.fitnass.data.model.Runs

@TypeConverters(ListConverter::class, DateTimeConverter::class)
@Database(entities = [Runs::class, ActiveRuns::class], version = 2, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {

    abstract fun runsDao(): RunsDao

    abstract fun activeRunsDao(): ActiveRunsDao

    companion object {
        private const val DB_NAME = "FitnassDatabase"
        private var INSTANCE: AppDatabase? = null

        fun getInstance(context: Context): AppDatabase {
            val temp = INSTANCE
            if (temp != null) {
                return temp
            }
            synchronized(this) {
                val instance = Room.databaseBuilder(context, AppDatabase::class.java, DB_NAME)
                        .fallbackToDestructiveMigration()
                        .allowMainThreadQueries()
                        .build()
                INSTANCE = instance
                return instance
            }
        }
    }
}